mlflow/store/artifact/databricks_artifact_repo.py

```diff
@@ -88,12 +89,30 @@ def _complete_futures(futures_dict):
     """
     results = {}
     errors = {}
-    for future in as_completed(futures_dict):
-        key = futures_dict[future]
-        try:
-            results[key] = future.result()
-        except Exception as e:
-            errors[key] = repr(e)
+
+    file_size = os.path.getsize(file)
+    pbar = tqdm(
+        total=file_size,
+        unit="iB",
+        unit_scale=True,
+        unit_divisor=1024,
+        desc=f"Uploading file {file}",
+        leave=False,
+    )
+    with pbar:
+        for index, future in enumerate(as_completed(futures_dict)):
+            key = futures_dict[future]
+            try:
+                results[key] = future.result()
+                pbar.update(
+                    min(
+                        file_size - index * _MULTIPART_UPLOAD_CHUNK_SIZE,
+                        _MULTIPART_UPLOAD_CHUNK_SIZE,
+                    )
+                )
+                pbar.refresh()
+            except Exception as e:
+                errors[key] = repr(e)
```

Comment:
I filed https://github.com/mlflow/mlflow/pull/8570. Once it's merged, we can pull out this logic from `_complete_futures`.